### PR TITLE
Option to override the ClinVar API URL to test against ClinVar 'apitest'

### LINKED ISSUE #4611: Prod/stage divide for CLINVAR settings
It would (sometimes) be useful to run tests on clinvar submission against the clinvar apitest. Moving variables like CLINVAR_API_URL to the server config file would make this more clear.

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Closes #4611

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Update stage server's [scout config file](https://github.com/Clinical-Genomics/servers/blob/master/config/CGVS/config/app/cg-services-stage.scilifelab.se/scout.conf.py) with this line:
CLINVAR_API_URL = "https://submit.ncbi.nlm.nih.gov/apitest/v1/submissions/" 
1. Restart scout stage
1. Stry to send a submission to ClinVar using a valid key. Make sure that response contains the URL of the apitest 

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/extensions/clinvar_extension.py [hunk 1]
```diff
@@ -15,9 +15,9 @@ class ClinVarApi:
     the ClinVar submission schema (https://www.ncbi.nlm.nih.gov/clinvar/docs/api_http/) and if valid, can be directly submitted to ClinVar.
     """
 
-    def __init__(self):
+    def init_app(self, app):
```
Comment (northwestwitch) [inline scout/server/extensions/clinvar_extension.py:18]: Has to be this way in order to be able to access the config file

File: scout/server/extensions/clinvar_extension.py [hunk 2]
```diff
@@ -15,9 +15,9 @@ class ClinVarApi:
...
         self.convert_service = "/".join([PRECLINVAR_URL, "csv_2_json"])
-        self.submit_service = CLINVAR_API_URL
+        self.submit_service = app.config.get("CLINVAR_API_URL") or CLINVAR_API_URL
```
Comment (dnil) [inline scout/server/extensions/clinvar_extension.py:20]: The name is good, but it is always a little confusing when the same name in different name spaces, that both are in use here. Consider something like `CLINVAR_API_URL_DEFAULT` for the const?

File: scout/server/extensions/clinvar_extension.py (hunk 1 above)
Comment (dnil) [reply scout/server/extensions/clinvar_extension.py:18]: All good!

File: scout/server/extensions/clinvar_extension.py (hunk 2 above)
Comment (dnil) [inline scout/server/extensions/clinvar_extension.py:20]:
```suggestion
        self.submit_service_url = app.config.get("CLINVAR_API_URL") or CLINVAR_API_URL_DEFAULT
```

**Review by dnil — APPROVED** [review]
Good to have! I'm afraid we will still be going a bit back and forth between the two for testing at different stages, but we should probably point servers stage config to `apitest` as a default after this.

File: scout/server/extensions/clinvar_extension.py (hunk 2 above)
Comment (northwestwitch) [reply scout/server/extensions/clinvar_extension.py:20]: Yes, good idea!

Comment (northwestwitch) [reply scout/server/extensions/clinvar_extension.py]: Fixed! I'm going to open a request in servers so We have the apitest in scout stage

**Pull request merged**